### PR TITLE
Javascript carousel - Vorleak Yek 

### PR DESCRIPTION
**Pull request by vorleakyek**
https://github.com/vorleakyek/m2-0723-code-solutions/assets/76607726/42508c8e-0473-4976-9a57-b9426635baa6

### CODE REVIEW

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    // change filled circle to open circle
+    showOpenCircles();
+  }, 3000);
+}
+
+// ************* Get updated async current index  **********************
+function getUpdatedValue(updatedValue) {
+  return updatedValue;
+}
+autoPlay(getUpdatedValue);
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js]: It seems like, the way `getUpdatedValue` is being used, it doesn't add any functionality.

File: javascript-carousel/main.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+nextButton.addEventListener('click', nextImg);
+previousButton.addEventListener('click', previousImg);
+
+function nextImg() {
+  clearInterval(intervalID);
+  currentIndex = currentIndex + 1 <= arrObj.length - 1 ? currentIndex + 1 : 0;
+  showImgAndFilledCircle();
+  showOpenCircles();
+  autoPlay(getUpdatedValue);
+}
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js]: `autoPlay` would be more self-contained if it handled clearing the previous interval internally.

File: javascript-carousel/main.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    return;
+  }
+  clearInterval(intervalID);
+  const currentCircleID = event.target.id;
+  currentIndex = arrObj.findIndex((obj) => obj.dotID === currentCircleID);
+  const allSpanCircles = document.querySelectorAll('li > span');
+  allSpanCircles.forEach((circle) => circle.classList.remove('filled-circle'));
+  showImgAndFilledCircle();
+  autoPlay(getUpdatedValue);
+}
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js]: I'd suggest tying `event.target`s and DOM queries to something more unique than `tagName`s. For example, by using classes.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by vorleakyek** [comment]
Okay, I made some changes based on the comments above. Please let make know if there is something that I should revise again.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**